pkg/kubelet/dockershim/docker_service.go

```diff
@@ -311,6 +319,89 @@ type dockerService struct {
 
 // TODO: handle context.
 
+// GetRuntimeConfigInfo returns the runtime config.
+func (ds *dockerService) GetRuntimeConfigInfo(_ context.Context, r *runtimeapi.GetRuntimeConfigInfoRequest) (*runtimeapi.GetRuntimeConfigInfoResponse, error) {
+	dockerInfo, err := ds.client.Info()
+	glog.Infof("Docker Info: %+v", dockerInfo)
+	if err != nil {
+		glog.Errorf("Failed to execute Info() call to the Docker client: %v", err)
+		return nil, err
+	}
+	uidMapping := &runtimeapi.LinuxIDMapping{ContainerId: uint32(0)}
+	gidMapping := &runtimeapi.LinuxIDMapping{ContainerId: uint32(0)}
+
+	var usernsEnabled bool
+	var remappedNonRootHostID uint32
+	for _, secOpt := range dockerInfo.SecurityOptions {
+		if strings.Contains(secOpt, "userns") {
+			usernsEnabled = true
+			break
+		}
+	}
+	if usernsEnabled {
+		if strings.HasPrefix(dockerInfo.DockerRootDir, "/var/lib/docker/") {
+			remappedNonRootHostID64, err := strconv.ParseUint(strings.Split(strings.TrimPrefix(dockerInfo.DockerRootDir, "/var/lib/docker/"), ".")[0], 10, 0)
+			if err != nil {
+				glog.Errorf("Failed to parse DockerRootDir: %v", err)
+				return nil, err
+			}
+			remappedNonRootHostID = uint32(remappedNonRootHostID64)
+			uidMapping.ContainerId = uint32(0)
+			gidMapping.ContainerId = uint32(0)
+
+			uidMapping.HostId = remappedNonRootHostID
+			gidMapping.HostId = remappedNonRootHostID
+		} else {
+			glog.Errorf("Failed in parsing DockerRootDir, %s, from Info(): %v", dockerInfo.DockerRootDir, err)
+			return nil, err
+		}
+		mappings, err := ioutil.ReadFile("/etc/subuid")
+		if err != nil {
+			glog.Errorf("Failed to read /etc/subuid: %v", err)
+			return nil, err
+		}
+		uidMapping.Size_, err = getIDMappingSize(mappings, remappedNonRootHostID)
+		if err != nil {
+			return nil, err
+		}
+
+		mappings, err = ioutil.ReadFile("/etc/subgid")
+		if err != nil {
+			glog.Errorf("Failed to read /etc/subgid: %v", err)
+			return nil, err
+		}
+		gidMapping.Size_, err = getIDMappingSize(mappings, remappedNonRootHostID)
+		if err != nil {
+			return nil, err
+		}
+	}
+	linuxConfig := &runtimeapi.LinuxUserNamespaceConfig{
+		IsEnabled:   usernsEnabled,
+		UidMappings: []*runtimeapi.LinuxIDMapping{uidMapping},
+		GidMappings: []*runtimeapi.LinuxIDMapping{gidMapping},
+	}
+	getRuntimeConfig := &runtimeapi.GetRuntimeConfig{UserNamespaceConfig: linuxConfig}
+	return &runtimeapi.GetRuntimeConfigInfoResponse{RuntimeConfig: getRuntimeConfig}, nil
+}
+
+func getIDMappingSize(mappings []byte, hostID uint32) (uint32, error) {
+	matches := linuxIDMappingRegexp.FindAllSubmatch(mappings, -1)
+	for _, match := range matches {
+		uid, err := strconv.ParseUint(string(match[2]), 10, 32)
+		if err != nil {
+			return 0, fmt.Errorf("Error in parsing linux user-namespace mapping entry: %s", match)
```

Comment:
errors should start with lower case `fmt.Errorf("error in parsing...")

make similar updates throughout.